### PR TITLE
ci: Create merge-gatekeeper.yml to make sure PR only merged when all tests checked.

### PR DESCRIPTION
**Pull request by noklam**
## Development notes

To make this work, it requires change of branch setting and add "Status Check" for `merge-gatekeeper`

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: .github/workflows/merge-gatekeeper.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Merge Gatekeeper
+
+on:
+  pull_request:
+    branches:
+      - main
+      - dev
```
Comment (merelcht) [inline .github/workflows/merge-gatekeeper.yml:7]: We don't have a `dev` branch, so I guess it's not needed to add it.

**Review by merelcht — APPROVED** [review]
LGTM! Thanks for finding this solution 👍

**Review by SajidAlamQB — APPROVED** [review]

File: .github/workflows/merge-gatekeeper.yml (hunk 1 above)
Comment (noklam) [reply .github/workflows/merge-gatekeeper.yml:7]: I change it to `develop` since we do make breaking change PR to `develop` from time to time.

Comment (noklam) [inline .github/workflows/merge-gatekeeper.yml:7]:
```suggestion
      - develop
```

**Pull request merged**